state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -944,13 +943,33 @@ impl<T: Config<I>, I: 'static> Broadcaster<T::TargetChain> for Pallet<T, I> {
 		(broadcast_id, Self::threshold_sign(api_call, broadcast_id, false))
 	}
 
-	fn threshold_resign(broadcast_id: BroadcastId) -> Option<ThresholdSignatureRequestId> {
-		ThresholdSignatureData::<T, I>::get(broadcast_id)
-			.map(|(api_call, _signature)| Self::threshold_sign(api_call, broadcast_id, false))
+	fn re_sign_broadcast(
+		broadcast_id: BroadcastId,
+		request_broadcast: bool,
+	) -> Result<ThresholdSignatureRequestId, DispatchError> {
+		AbortedBroadcasts::<T, I>::mutate(|aborted| {
+			aborted.remove(&broadcast_id);
+		});
+		let api_call = Self::clean_up_broadcast_storage(broadcast_id)
+			.ok_or(Error::<T, I>::ApiCallUnavailable)?;
+
+		PendingBroadcasts::<T, I>::try_mutate(|pending| {
+			if pending.contains(&broadcast_id) {
+				Err(Error::<T, I>::BroadcastStillPending)
+			} else {
+				if request_broadcast {
+					pending.insert(broadcast_id);
+				}
+				Ok(())
+			}
+		})?;
+		Ok(Self::threshold_sign(api_call, broadcast_id, request_broadcast))
 	}
 
-	/// Clean up storage data related to a broadcast ID.
-	fn clean_up_broadcast_storage(broadcast_id: BroadcastId) {
+	fn expire_broadcast(broadcast_id: BroadcastId) {
+		// These would otherwise be cleaned up when the broadacst succeeds or aborts.
+		RequestSuccessCallbacks::<T, I>::remove(broadcast_id);
+		RequestFailureCallbacks::<T, I>::remove(broadcast_id);
 		Self::clean_up_broadcast_storage(broadcast_id);
 	}
```

Comment:
maybe we should call the ```clean_up_broadcast_storage``` in abort_broadcast instead of calling separately in ```re_sign_broadcast``` and ```expire_broadcast``` which should be equivalent. This would be more readable conceptually since the ```clean_up_broadcast_storage``` clears all the broadcast related storages which become irrelevant when we abort (since we dont broadcast anymoe).

Reply:
If we call it in `abort_broadcast` then we discard the api-call, which we need in order to re-sign.

Reply:
hm.. we could not delete the ```ThresholdSignatureData``` storage in ```clean_up_broadcast_storage``` which contains the apicall. We could then delete it in the expiry and broadcast success. ```ThresholdSignatureData``` is actually just a stored apicall for each broadcast_id. The signature field is not used (see the other comment). It should probably be renamed to something like ```Apicalls```. This way (not deleting the apicall storage and calling cleanup in abort broadcast) feels more intuitive to me but I guess it is a matter of choice so I am ok if you dont want to change it.

Reply:
I can see your point, will rename and see if I think it makes sense to restructure further.

Reply:
I did the renaming.

I also had another look and I don't think one approach is clearly better than the other: when we succeed we want to remove everything, but when we abort we want to keep the api call. So either way there needs to be some flag or special case in the cleanup. Will leave it as is.